src/sql/platform/connection/common/connectionProfile.ts

```diff
@@ -84,30 +83,14 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 	}
 
 	public static matchesProfile(a: interfaces.IConnectionProfile, b: interfaces.IConnectionProfile): boolean {
-		return a && b
-			&& a.providerName === b.providerName
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.serverName, b.serverName)
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.databaseName, b.databaseName)
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.userName, b.userName)
-			&& ConnectionProfile.nullCheckEqualsIgnoreCase(a.options['databaseDisplayName'], b.options['databaseDisplayName'])
-			&& a.authenticationType === b.authenticationType
-			&& a.groupId === b.groupId;
+		return a && b && a.getOptionsKey() === b.getOptionsKey();
```

Comment:
If these can be undefined please update the param to be `interfaces.IConnectionProfile | undefined`